app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js

```diff
@@ -1,415 +1,355 @@
-       expect(result).toEqual({ dirty: { securityGroups: ['sg1', 'sg2'] }});
-     });
-
-     it('updates filteredData to new region - no VPC', function() {
-       var expected = this.allSecurityGroups.test.azure['us-east-1'].slice(0, 2);
-       this.command.region = 'us-east-1';
-       service.configureSecurityGroupOptions(this.command);
-       expect(this.command.backingData.filtered.securityGroups).toEqual(expected);
-     });
-
-     it('updates filteredData to new VPC', function() {
-       var expected = this.allSecurityGroups.test.azure['us-west-1'].slice(3, 5);
-       this.command.vpcId = 'vpc-1';
-       service.configureSecurityGroupOptions(this.command);
-       expect(this.command.backingData.filtered.securityGroups).toEqual(expected);
-     });
-
-   });
-
-   describe('configureKeyPairs', function() {
-     beforeEach(function() {
-       this.command = {
-         backingData: {
-           filtered: {},
-           credentialsKeyedByAccount: {
-             test: {
-               defaultKeyPair: 'test-pair'
-             },
-             prod: {
-               defaultKeyPair: 'prod-pair'
-             },
-             nothing: {}
-           },
-           keyPairs: [
-             { account: 'test', region: 'us-west-1', keyName: 'test-pair' },
-             { account: 'test', region: 'us-east-1', keyName: 'test-pair' },
-             { account: 'test', region: 'us-west-1', keyName: 'shared' },
-             { account: 'prod', region: 'us-west-1', keyName: 'prod-pair' },
-             { account: 'prod', region: 'us-west-1', keyName: 'shared' },
-           ]
-         },
-         credentials: 'test',
-         region: 'us-west-1',
-         keyPair: 'shared',
-       };
-     });
-
-     it('retains keyPair when found in new account', function() {
-       this.command.credentials = 'prod';
-       service.configureKeyPairs(this.command);
-       expect(this.command.keyPair).toBe('shared');
-     });
-
-     it('retains keyPair when found in new region', function() {
-       this.command.region = 'us-east-1';
-       this.command.keyPair = 'test-pair';
-       service.configureKeyPairs(this.command);
-       expect(this.command.keyPair).toBe('test-pair');
-     });
-
-     it('marks dirty, sets to new default value when key pair not found in new account', function() {
-       this.command.credentials = 'prod';
-       this.command.keyPair = 'test-pair';
-       var result = service.configureKeyPairs(this.command);
-       expect(result.dirty.keyPair).toBe(true);
-       expect(this.command.keyPair).toBe('prod-pair');
-     });
-
-     it('marks dirty, sets to default value when key pair not found in new region', function() {
-       this.command.region = 'us-east-1';
-       var result = service.configureKeyPairs(this.command);
-       expect(result.dirty.keyPair).toBe(true);
-       expect(this.command.keyPair).toBe('test-pair');
-     });
-
-   });
-
- });
+/*
+ * Copyright 2014 Netflix, Inc.
```

Comment:
Consider changing "2014 Netflix" to "2016 Microsoft" - or removing the header altogether.